src/tests/test_bootstrap.py

```diff
@@ -0,0 +1,47 @@
+"""
+Separate test case for bootstrapping - connecting to network
+with stale knownnodes.
+"""
+
+import json
+import os
+import tempfile
+import time
+
+from test_api import TestAPIProto
+
+
+class TestBootstrap(TestAPIProto):
+    """Test case for bootstrapping"""
+    @classmethod
+    def setUpClass(cls):
+        knownnodes_file = os.path.join(  # cls.home is not set ):
+            tempfile.gettempdir(), 'knownnodes.dat')
+        with open(knownnodes_file, 'wb') as dst:
+            json.dump([
+                {  # One of DEFAULT_NODES, so knownNodesActual is False
+                    "peer": {
+                        "host": "158.222.217.190",
+                        "port": 8080
+                    },
+                    "info": {
+                        "rating": 0.0,
+                        "self": False,
+                        "lastseen": time.time()
+                    },
+                    "stream": 1
+                }
+            ], dst)
+        cls._started = time.time()
+        super(TestBootstrap, cls).setUpClass()
+
+    def test_connected(self):
+        """Ensure that pybitmessage connects to network in 5 min"""
+        for _ in range(60):
```

Comment:
Maybe a different way to check time in case the API call takes too long. And what if the API call freezes, shouldn't there be checks for that as well?

Reply:
It's just a first try. This test not actually tests what I need ):